backpropagation/network/neural_network.py

```diff
@@ -41,16 +41,34 @@ def __init__(
         self.layers_count = len(neurons_count_per_layer)
 
         self.weights = [
-            np.random.normal(size=(
+            self._create_weight_matrix(size=(
                 neurons_count_per_layer[layer + 1],
-                neurons_count_per_layer[layer] + 1
+                neurons_count_per_layer[layer]
             ))
             for layer in range(self.layers_count - 1)
         ]
-
         self.activation_function = activation_function
         self.cost_function = cost_function
 
+    def _create_weight_matrix(self, size: Tuple[int, int]) -> np.array:
+        """Creates a weight matrix between layers of sizes specified by ``size``.
+
+            It creates the biases weights and stacks it horizontally in front of the
+            weight matrix.
+
+            Args:
+                size (Tuple[int, int]): Size of the layers between which the matrix
+                should be initialized.
+
+            Returns (np.array): Weight matrix used to calculate the activation from one
+                layer to another.
+        """
+        rows, columns = size
+        biases = np.random.normal(size=(rows, 1))
+        weight_matrix = np.random.normal(size=(rows, columns), scale=1/np.sqrt(columns))
```

Comment:
What is the reason behind this specific scale parameter?

Reply:
You make the distribution "flatter" and by that neurons are less likely to saturate. You can read more on that here: http://neuralnetworksanddeeplearning.com/chap3.html#weight_initialization